source/coap_connection_handler.c

```diff
@@ -596,8 +596,7 @@ void connection_handler_destroy(coap_conn_handler_t *handler)
             }
         }
         int_socket_delete(handler->socket);
-        handler->socket = NULL;
-        ns_dyn_mem_free(handler);
+
```

Comment:
ns_dyn_mem_free(handler); should be called here, and inside if(handler) - braces . Mika have refactored this function, so I don't this is needed.